api/bertymessenger.proto

```diff
@@ -264,18 +264,18 @@ message Account {
 }
 
 message Interaction {
-  string cid = 1 [(gogoproto.moretags) = "gorm:\"uniqueIndex\"", (gogoproto.customname) = "CID"];
-  AppMessage.Type type = 2;
+  string cid = 1 [(gogoproto.moretags) = "gorm:\"uniqueIndex\"", (gogoproto.customname) = "CID"]; // FIXME(?): primaryKey instead of uniqueIndex?
+  AppMessage.Type type = 2; // FIXME(?): TypeUserReaction, TypeAcknowledge -> no targetCID currently exists, should we add this to this model, could be great, add index to field?
   string member_public_key = 7;
   string device_public_key = 12;
   Member member = 8;
-  string conversation_public_key = 3;
+  string conversation_public_key = 3; // FIXME(?): add index
   Conversation conversation = 4;
   bytes payload = 5;
   bool is_me = 6;
   int64 sent_date = 9;
-  bool acknowledged = 10;
-  int64 arrival_index = 11 [(gogoproto.moretags) = "gorm:\"autoIncrement\""];
+  bool acknowledged = 10; // FIXME(?): By whom? Would a timestamp be better?
+  int64 arrival_index = 11 [(gogoproto.moretags) = "gorm:\"autoIncrement\""]; // FIXME(?): I think this implies a primary key field, can't we set a timestamp instead?
```

Comment:
yes but as you pointed system clocks can be off.
this isz only used to get the "acks/messages without device" backlog in the correct order